_rules/SC4-1-2-role-has-required-states-and-properties.md

```diff
@@ -0,0 +1,131 @@
+---
+name: Role has required states and properties
+description: |
+  Elements that has a semantic role must also specify all required states and properties
+  
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+Any HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies) and has an [explicit semantic role](#semantic-role), except if the element has an [implicit semantic role](#implicit-role) that is identical to the explicit semantic role. 
+
+### Expectation
+
+For each test target all attributes listed under [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) in [WAI-ARIA](https://www.w3.org/TR/wai-aria) for that role are present and [non-empty](#non-empty), unless they have a default value listed under [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole).
```

Comment:
This can be simplified quite a bit:
- Make the "non-empty" thing its own expectation
- Make the expectation something like: "For each test target, the required attributes for the role is present, unless the attribute has a default value. The list of required attributes for a role can be found in ARIA 1.1" That breaks it up, makes it easier to read.